api/phone.go

```diff
@@ -29,39 +34,51 @@ func (a *API) formatPhoneNumber(phone string) string {
 	return strings.ReplaceAll(strings.Trim(phone, "+"), " ", "")
 }
 
-func (a *API) sendPhoneConfirmation(ctx context.Context, tx *storage.Connection, user *models.User, phone string) error {
+// sendPhoneConfirmation sends an otp to the user's phone number
+func (a *API) sendPhoneConfirmation(ctx context.Context, tx *storage.Connection, user *models.User, phone, otpType string, smsProvider sms_provider.SmsProvider) error {
```

Comment:
Changed the function signature here to take in an `otpType` and a `smsProvider`. The `smsProvider` was added to the function params to make it easier to test this function with a mock sms provider that implements a stub `SendSms` method. See [TestSendPhoneConfirmation](https://github.com/supabase/gotrue/pull/421/files#diff-10c35f776118e94548a7f2835c538ba3996e74fdf24df35e5591922bad18d8b9R66)